pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -152,6 +155,34 @@ func (rs *RevisionSpec) Validate(ctx context.Context) *apis.FieldError {
 	return errs
 }
 
+func validateAnnotations(annotations map[string]string) *apis.FieldError {
+	return validatePercentageAnnotationKey(annotations, serving.QueueSideCarResourcePercentageAnnotation)
+}
+
+func validatePercentageAnnotationKey(annotations map[string]string, resourcePercentageAnnotationKey string) *apis.FieldError {
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	v, ok := annotations[resourcePercentageAnnotationKey]
+	if !ok {
+		return nil
+	}
+	value, err := strconv.ParseFloat(v, 32)
+	if err != nil {
+		return &apis.FieldError{
+			Message: fmt.Sprintf("Invalid value %s for annotation %s: %v", v, resourcePercentageAnnotationKey, err),
+			Paths:   []string{resourcePercentageAnnotationKey},
+		}
+	}
+
+	if value <= float64(0) || value > float64(100) {
+		return apis.ErrOutOfBoundsValue(value, 0.1, 100.0, resourcePercentageAnnotationKey)
+	}
```

Comment:
mismatch between 0.1 (here) and 0 (above)?

I recently changed the signature of this to allow arbitrary types through, so it's neat to see you using it for floats! :)